router/customdestinationmanager/customdestinationmanager_test.go

```diff
@@ -197,3 +269,154 @@ func TestKVManagerInvocations(t *testing.T) {
+		ID: "someDestinationID1",
+		DestinationDefinition: backendconfig.DestinationDefinitionT{
+			Name: "REDIS",
+		},
+	}
+	err := customManager.onNewDestination(someDestination)
+	assert.Nil(t, err)
+
+	for _, tc := range redisJSONTestCases {
+		t.Run(tc.description, func(t *testing.T) {
+			redisAddr, destroy := testutils.StartRedis(t, tc.redisImgRepo, tc.redisImgTag)
+			defer destroy()
+			event, err := json.Marshal(tc.transformedResponse)
+			if err != nil {
+				t.Errorf("MarshalError: %v\n", err.Error())
+				return
+			}
+			config := map[string]interface{}{
+				"useJSONModule": true,
+				"address":       redisAddr,
+				"db":            0,
+				"clusterMode":   false,
+			}
+			kvMgr := &kvstoremanager.RedisManagerT{
+				Config: config,
+			}
+			kvMgr.Connect()
+			db := kvMgr.GetClient()
+
+			stCd, er := customManager.send(event, kvMgr, config)
+			if er != "" {
+				t.Logf("Error: %s\n", er)
+				require.Contains(t, er, tc.expectedSendDataResponse.err)
+				return
+			}
+			msgMap := gjson.GetBytes(event, "message.value").String()
+			key := gjson.GetBytes(event, "message.key").String()
+			res, err := db.JSONGet(context.TODO(), key).Result()
+			if err != nil {
+				t.Error(err)
+				return
+			}
+			require.Equal(t, msgMap, res)
+
+			require.Equal(t, tc.expectedSendDataResponse.statusCode, stCd)
+		})
+	}
+}
+
+func TestRedisMgrForMultipleJSONsSameKey(t *testing.T) {
+	initCustomerManager()
+	customManager := New("REDIS", Opts{}).(*CustomManagerT)
+	someDestination := backendconfig.DestinationT{
+		ID: "someDestinationID1",
+		DestinationDefinition: backendconfig.DestinationDefinitionT{
+			Name: "REDIS",
+		},
+	}
+	err := customManager.onNewDestination(someDestination)
+	assert.Nil(t, err)
+
+	transformedResponse := transformedResponseJSON{
+		Message: map[string]interface{}{
+			"key":  "user:myuser-id",
+			"path": "mode-1",
+			"value": map[string]interface{}{
+				"key": "someKey",
+				"fields": map[string]interface{}{
+					"field1": "value1",
+					"field2": 2,
+				},
+			},
+		},
+		UserId: "myuser-id",
+	}
+
+	t.Run("When JSON module is loaded into Redis, path is mentioned and key is not present in Redis, should set into redis and fetching the data should be successful", func(t *testing.T) {
+		redisAddr, destroy := testutils.StartRedis(t, "redis/redis-stack-server", "latest")
+		defer destroy()
+		event, err := json.Marshal(transformedResponse)
+		if err != nil {
+			t.Errorf("MarshalError: %v\n", err.Error())
+			return
+		}
+		config := map[string]interface{}{
+			"useJSONModule": true,
+			"address":       redisAddr,
+			"db":            0,
+			"clusterMode":   false,
+		}
+		kvMgr := &kvstoremanager.RedisManagerT{
+			Config: config,
+		}
+		kvMgr.Connect()
+		db := kvMgr.GetClient()
+
+		stCd, _ := customManager.send(event, kvMgr, config)
+		v, err := db.JSONGet(context.TODO(), "user:myuser-id", "$.mode-1").Result()
+
+		require.Nil(t, err)
+		require.NotEqual(t, "", v)
```

Comment:
We don't need this assertion, since below we are verifying for a expected string.